core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/auth/ModelAuthProvider.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkiverse.langchain4j.runtime.auth;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import jakarta.enterprise.inject.Instance;
+import jakarta.enterprise.inject.spi.CDI;
+
+import io.quarkiverse.langchain4j.ModelName;
+
+public interface ModelAuthProvider {
+    String getAuthorization(Input input);
+
+    interface Input {
+        String method();
+
+        URI uri();
+
+        Map<String, List<Object>> headers();
+    }
+
+    class Resolver {
```

Comment:
Actually, we don't even need this class, we can just add the `resolve()` methods as static methods to `ModelAuthProvider`